pennylane/ops/identity.py

```diff
@@ -57,7 +57,8 @@ def _matrix(cls, *params):
     def _heisenberg_rep(p):
         return np.array([1, 0, 0])
 
-    def diagonalizing_gates(self):
+    @staticmethod
+    def compute_diagonalizing_gates(*params, wires, **hyperparams):
```

Comment:
We could use the signature `compute_diagonalizing_gates()` here, but then we'd have to also overwrite `diagonalizing_gates` which calls the method. I think this is neater, even though the arguments don't get used?

Reply:
Agree!